src/adaptors/tcp/tcp_adaptor.c

```diff
@@ -2387,6 +2409,20 @@ QD_EXPORT void qd_dispatch_delete_tcp_listener(qd_dispatch_t *qd, void *impl)
             listener->common.vflow = 0;
         }
         //
+        // Initiate termination of existing connections
+        //
+        if (tcp_context->qd->terminate_tcp_conns) {
+            // Note: PN_RAW_CONNECTION_CONNECTED event or PN_RAW_CONNECTION_DISCONNECTED event
+            // could come upon any of the connections. We need to hold the listener->lock
+            // to prevent any modification of the connections list while it is being traversed.
```

Comment:
Should probably explain why we are only waking the head connection rather than all of them.